src/server.py

```diff
@@ -78,6 +81,37 @@ def get_date_range(dates):
                 'endDate': max(dates).strftime('%Y-%m-%d %H:%M')}
 
 
+def valve(head_rev, pushlog_id):
+    """Determine which kind of job should been returned"""
+    low_job_return = True
+    COUNTER.increase_the_counter()
+    request_list = []
+    try:
+        request_list = session.query(TaskRequests.head_rev, TaskRequests.pushlog_id,
+                                     TaskRequests.low_job_return).all()
+    except Exception:
+        session.rollback()
+
+    requests = {}
+    for head_rev, pushlog_id, low_job_return in request_list:
+        requests[pushlog_id] = {'head_rev': head_rev,
+                                'low_job_return': low_job_return}
+    # If this pushlog_id has been schduled, we just return
+    # the priority returned before.
+    if pushlog_id in requests.keys():
+        low_job_return = requests.get(pushlog_id)['low_job_return']
+    else:
+        # we return all jobs for every 5 pushes.
+        if COUNTER.counter >= 5:
+            COUNTER.reset()
+            low_job_return = False
+        task_request = TaskRequests(str(head_rev), str(pushlog_id), low_job_return)
+        session.add(task_request)
+        session.commit()
+
+    return low_job_return
```

Comment:
we have discussed high value vs low value for the jobs, in this case I see that:
* low_job_return == True: high value jobs
* low_job_return == False: all jobs (high value + low value)

if we think about priority as multiple levels of job values we support.  Ideally we would use a "reverse" priority system where:
priority 0 == ALL JOBS (currently low value + high value)
priority 1 == high value jobs

in the future we could have:
priority 0 == ALL JOBS
priority 1 == jobs which have found at least 1 failure in the past
priority 2 == jobs which have found at least 2 failures in the past
priority 3 == jobs which have found at least 3 failures in the past
priority 4 == jobs which have found at least 4 failures in the past

then we could determine a skipcount for each priority- as in:
every 1 push = priority 4
every 2 push = priority 3
every 3 push = priority 2
every 4 push = priority 1
every 5 push = priority 0

or more simply put- the counter that we have would start at 5 and decrement to 0, when we hit 0 reset to 5 and we just run the query of jobs from the table which match the counter :)